foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/platform/database/Oracle8Platform.java

```diff
@@ -14,12 +14,274 @@
+        super.copyInto(platform);
+        if (!(platform instanceof Oracle8Platform)) {
+            return;
+        }
+        Oracle8Platform oracle8Platform = (Oracle8Platform) platform;
+        oracle8Platform.setShouldUseLocatorForLOBWrite(shouldUseLocatorForLOBWrite());
+        oracle8Platform.setLobValueLimits(getLobValueLimits());
+    }
+
+    /**
+     * INTERNAL Used by SQLCall.appendModify(..) If the field should be passed
+     * to customModifyInDatabaseCall, retun true, otherwise false. Methods
+     * shouldCustomModifyInDatabaseCall and customModifyInDatabaseCall should be
+     * kept in sync: shouldCustomModifyInDatabaseCall should return true if and
+     * only if the field is handled by customModifyInDatabaseCall.
+     */
+    @Override
+    public boolean shouldUseCustomModifyForCall(DatabaseField field) {
+        if (shouldUseLocatorForLOBWrite()) {
+            Class type = field.getType();
+            if (ClassConstants.BLOB.equals(type) || ClassConstants.CLOB.equals(type)) {
+                return true;
+            }
+        }
+        return super.shouldUseCustomModifyForCall(field);
+    }
+
+    /**
+     * INTERNAL: Return if the LOB value size is larger than the limit, i.e. 4k.
+     */
+    protected boolean lobValueExceedsLimit(Object value) {
+        if (value == null) {
+            return false;
+        }
+        int limit = getLobValueLimits();
+        if (value instanceof byte[]) {// blob
+            return ((byte[]) value).length >= limit;
+        } else if (value instanceof String) {// clob
+            return ((String) value).length() >= limit;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * INTERNAL: This method is used to unwrap the oracle connection wrapped by
+     * the application server. TopLink needs this unwrapped connection for
+     * certain Oracle Specific support. (ie TIMESTAMPTZ, LOB) This is added as a
+     * workaround for bug 4565190
+     */
+    @Override
+    public Connection getConnection(AbstractSession session, Connection connection) {
+        if (session.getServerPlatform() != null && (session.getLogin()).shouldUseExternalConnectionPooling()) {
+            // This is added as a workaround for bug 4460996
+            return session.getServerPlatform().unwrapConnection(connection);
+        }
+        return connection;
+    }
+
+    /**
+     * INTERNAL Used by SQLCall.translate(..) Typically there is no field
+     * translation (and this is default implementation). However on different
+     * platforms (Oracle) there are cases such that the values for binding and
+     * appending may be different (BLOB, CLOB). In these special cases the
+     * method returns a wrapper object which knows whether it should be bound or
+     * appended and knows how to do that.
+     */
+    @Override
+    public Object getCustomModifyValueForCall(Call call, Object value, DatabaseField field, boolean shouldBind) {
```

Comment:
org.eclipse.persistence.platform.database.oracle.Oracle9Platform overrides this method as well, but since it calls super.getCustomModifyValueForCall(...), it might not be important to port. This goes with the same comment I have about if we need to add NClob support... I'm fine either way, just pointing it out.

Reply:
noted.